src/code_43F0.c

```diff
@@ -193,31 +218,27 @@ s32 func_80029994(s32 arg0) {
     return result;
 }
 
-// Issue with the negation at the beginning
-#ifdef NON_MATCHING
 s32 rand_int(s32 arg0) {
-    s32 ret;
+    s32 ret = 0;
 
-    if (arg0 < 0) {
-        arg0 = -arg0;
-    }
+    arg0 = abs(arg0);
 
-    ret = 0;
     if (arg0 != 0) {
-        if (arg0 == 1) {
-            return (func_80029994(1000) < 501) ^ 1;
+        switch (arg0) {
+            case 1:
+                ret = (func_80029994(1000) < 501) == 0;
```

Comment:
Use `!`?

Reply:
Thanks, I re-looked at this and realized I can do

```
            case 1:
                ret = func_80029994(1000) > 500;
                break;
```
which I think is a lot better